### DISCUSSION TITLE
Adding roles in a different guild

### DISCUSSION THREAD

**Discussion by aayushkdev** [Rapptz/discord.py#5903] [Q&A]
I want to add a role in one server and the role should be added in the other server is there anyway I can do it

**Comment by Rapptz** [comment]
Roles are server-specific. I'm assuming you mean you want to clone a role from guild A to guild B.

To do this, you just create a role in the guild you care about with the same attributes, for example:

```py
first_guild_id = 123 # The guild ID of the guild with the role you want to clone
first_guild = bot.get_guild(first_guild_id)

second_guild_id = 123 # The guild ID of the guild you want to clone the role to
second_guild = bot.get_guild(second_guild_id)

role_id = 123 # The role ID of the role you want to clone
role = first_guild.get_role(role_id)

fields = {
    'name': role.name,
    'permissions': role.permissions,
    'colour': role.colour,
    'hoist': role.hoist,
    'mentionable': role.mentionable,
}
await second_guild.create_role(**fields, reason=f'Cloning role from {first_guild.name}')
```

Note that IDs are integers. If it's not found then it'll return `None`. Likewise, you need permissions to create a role in the second guild.

**Comment by aayushkdev** [comment]
no see there are two servers one is staff and one is the main so if I run the command in one server it adds the role in both servers to a specific member (the roles are already created)

**Reply by Rapptz** [reply]
Well, in this case you're better off posting what you tried.

**Reply by aayushkdev** [reply]
```py
import discord.member
import discord
from discord.ext import commands

class Roles(commands.Cog):
    def __init__(self, bot):
        self.bot = bot


    @commands.command()
    async def test(self, ctx, memeber : discord.Member):
        role = discord.utils.get(self.bot.guilds[1].roles, id=763047279912419348)
        await memeber.add_roles(role)
        

def setup(bot):
    bot.add_cog(Roles(bot))
```
this is what I did till now

**Reply by Rapptz** [reply]
Then no, this isn't possible. A role is server specific. You can't add a role that exists in server A to a member in server B.

**Reply by aayushkdev** [reply]
ok is it possible to add the role to the second server to a specific person from server 1 
for example I am in server 2 and you run a command in server 1 so it adds the role to me in the other server(the role exist in server 2)

**Reply by Rapptz** [reply]
It's the same steps as the answer above instead you use [`Member.add_roles`](https://discordpy.readthedocs.io/en/latest/api.html#discord.Member.add_roles) instead of create_roles.

**Reply by aayushkdev** [reply]
is it something like this
```py
@commands.command()
    async def add(self, ctx, member : discord.Member):
        first_guild_id = 750959329464549466 # The guild ID of the guild with the role you want to clone
        first_guild = self.bot.get_guild(first_guild_id)

        second_guild_id = 763041558014066699 # The guild ID of the guild you want to clone the role to
        second_guild = self.bot.get_guild(second_guild_id)
        role = discord.utils.get(self.bot.guilds[1].roles, name="members")
        await member.add_role(role)
```


I am getting discord.ext.commands.errors.CommandInvokeError: Command raised an exception: AttributeError: 'Member' object has no attribute 'add_role'

**Reply by Rapptz** [reply]
Notice I said add_role**s**. Also, don't do `self.bot.guilds[1]`, you already have the guild objects in the line above.

Anyway, the `member` object you have passed will be the member for the guild of where the command is not the one in the other guild.

**Reply by aayushkdev** [reply]
so how should I change the member object to make it of the other guild
